kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -360,143 +360,9 @@ bool FindIntersectedGeometricalObjectsProcess::HasIntersection(
-        }
-    } else {
-        if (local_space_dimension == 3) {
-            return this->HasIntersection3D(rFirstGeometry, rSecondGeometry);
-        } else {
-            return this->HasDirectIntersection3D(rFirstGeometry, rSecondGeometry);
-        }
-    }
-}
-
-/***********************************************************************************/
-/***********************************************************************************/
-
-bool FindIntersectedGeometricalObjectsProcess::HasIntersection2D(
-    GeometryType& rFirstGeometry,
-    GeometryType& rSecondGeometry
-    )
-{
-    // Check the intersection of each edge against the intersecting object
-    const array_1d<double, 3>& r_coordinates_second_geometry_1 = rSecondGeometry[0].Coordinates();
-    const array_1d<double, 3>& r_coordinates_second_geometry_2 = rSecondGeometry[1].Coordinates();
-    const auto edges = rFirstGeometry.GenerateEdges();
-    PointType int_pt(0.0,0.0,0.0);
-    for (auto& edge : edges) {
-        const int int_id = IntersectionUtilities::ComputeLineLineIntersection<Line2D2<NodeType>>(
-            Line2D2<NodeType>{edge},
-            r_coordinates_second_geometry_1,
-            r_coordinates_second_geometry_2,
-            int_pt.Coordinates());
-
-        if (int_id != 0){
-            return true;
-        }
-    }
-
-    // Let check second geometry is inside the first one.
-    // Considering that there are no intersection, if one point is inside all of it is inside.
-    array_1d<double, 3> local_point;
-    if (rFirstGeometry.IsInside(rSecondGeometry.GetPoint(0), local_point)){
-        return true;
-    }
-
-    return false;
-}
-
-/***********************************************************************************/
-/***********************************************************************************/
-
-bool FindIntersectedGeometricalObjectsProcess::HasDirectIntersection2D(
-    GeometryType& rFirstGeometry,
-    GeometryType& rSecondGeometry
-    )
-{
-    // Check the intersection of each edge against the intersecting object
-    const array_1d<double, 3>& r_coordinates_second_geometry_1 = rSecondGeometry[0].Coordinates();
-    const array_1d<double, 3>& r_coordinates_second_geometry_2 = rSecondGeometry[1].Coordinates();
-    PointType int_pt(0.0,0.0,0.0);
-    const int int_id = IntersectionUtilities::ComputeLineLineIntersection<Line2D2<NodeType>>(
-        Line2D2<NodeType>{rFirstGeometry},
-        r_coordinates_second_geometry_1,
-        r_coordinates_second_geometry_2,
-        int_pt.Coordinates());
-
-    if (int_id != 0){
-        return true;
-    }
-
-    // Let check second geometry is inside the first one.
-    // Considering that there are no intersection, if one point is inside all of it is inside.
-    array_1d<double, 3> local_point;
-    if (rFirstGeometry.IsInside(rSecondGeometry.GetPoint(0), local_point)){
-        return true;
-    }
-
-    return false;
+    return rFirstGeometry.HasIntersection(rSecondGeometry);
```

Comment:
I think that now we can remove this `HasIntersection` from the `FindIntersectedGeometricalObjectsProcess`. It implements one line and (as far as I see) it is used once.

Reply:
It is override in `FindIntersectedGeometricalObjectsWithOBBProcess`

Reply:
OK. Hence this is related to my question below, isn't possible to use the intersection utilities in the OBB find intersected geometrical objects process?

Reply:
mmmh, I'm taking a look into `oriented_bounding_box.h`